src/views/homepage/components/product-nav/product-nav.module.css

```diff
@@ -0,0 +1,132 @@
+.productNav {
+  --container-max-width: 1240px;
+  --padding-x: 24px;
+  --notice-background-color: var(--token-color-palette-neutral-700);
+
+  position: relative;
+}
+
+.notice {
+  position: relative;
+  margin: 0;
+  padding: 0 var(--padding-x) 12px;
+  background-color: var(--notice-background-color);
+  text-align: center;
+}
+
+.noticeText {
+  margin: 0;
+  padding: 0;
+  color: var(--token-color-palette-neutral-400);
+}
+
+.nav {
+  position: relative;
+  padding: 0 var(--padding-x);
+  background-image: linear-gradient(
+    to bottom,
+    var(--notice-background-color) 24px,
+    var(--token-color-palette-neutral-0) 24px
+  );
+}
+
+.list {
+  list-style: none;
+  margin: 0 auto;
+  padding: 0;
+  max-width: var(--container-max-width);
+  display: grid;
+  grid-template-columns: repeat(2, 1fr);
+  gap: 24px;
+  border-radius: 14px;
+
+  @media (--dev-dot-tablet-up) {
+    grid-template-columns: repeat(3, 1fr);
+  }
+
+  @media (--dev-dot-desktop) {
+    display: flex;
+    gap: 0;
+    box-shadow: var(--token-surface-base-box-shadow);
+  }
+}
+
+.listItem {
+  position: relative;
+  flex: 1;
+}
+
+.product {
+  position: relative;
+  display: grid;
+  place-items: center;
+  padding-top: 24px;
+  padding-bottom: 24px;
+  border-radius: 14px;
+  background-color: var(--token-color-surface-primary);
+  color: var(--token-color-foreground-disabled);
+  box-shadow: var(--token-surface-base-box-shadow);
+  overflow: hidden;
+
+  @media (--dev-dot-desktop) {
+    box-shadow: none;
+    border-radius: 0;
+    overflow: visible;
+
+    @nest .listItem:first-child & {
+      border-top-left-radius: 14px;
+      border-bottom-left-radius: 14px;
+
+      &::after, &:hover::after {
+        border-bottom-left-radius: 14px;
+      }
+    }
+
+    @nest .listItem:last-child & {
+      border-top-right-radius: 14px;
+      border-bottom-right-radius: 14px;
+
+      &::after, &:hover::after {
+        border-bottom-right-radius: 14px;
+      }
+    }
```

Comment:
Oooh I think this is the first time I've seen `@nest`! 🤩 

Nitpick: I find `@nest` stuff to be a bit confusing - and it feels like it might break our convention of [grouping styles by selector](https://www.notion.so/hashicorp/CSS-Modules-753f2647f7aa4a598e28990d9f0d05a8).

Definitely optional, but in the interest of keeping styles grouped by selector, would it make sense here to use `index` in the `products` map to generate & pass `isFirstItem` & `isLastItem` classNames rather than only `s.product`? This way we could avoid the parent-dependent `@nest` selector:


```jsx
/* products.map((product, index: number) => { */

const productClassName = classNames(s.product, {
  [s.isFirstChild]: index == 0,
  [s.isLastChild]: index == products.length - 1,
})

/* ... and pass `productClassName` instead of `s.product` ... */
```

```suggestion
    &.isFirstChild {
      border-top-left-radius: 14px;
      border-bottom-left-radius: 14px;

      &::after, &:hover::after {
        border-bottom-left-radius: 14px;
      }
    }

    &.isLastChild {
      border-top-right-radius: 14px;
      border-bottom-right-radius: 14px;

      &::after, &:hover::after {
        border-bottom-right-radius: 14px;
      }
    }
```

Reply:
Got it, I am on board with this change.

I tend to prefer to use nest as to reduce the code needed within the component, since this can be done in CSS without the extra code. This is a bit trivial since its just two classes but on a component when more children need to change based on a state or position of index I'd prefer not to have to add conditional classes throughout the component when CSS can handle it. I see these two variations one of the same with regards to grouping, these two selectors are at the same level, we're offloading more work to the component code.

Reply:
☝️ Great point about reducing the code & # of tacked-on classNames within the component - I can definitely see the rationale for that.

Also 🤦 on mentioning grouping, I had a more convoluted idea in mind (see `Random related thoughts`) but didn't go through with it - very much not applicable in this case

I should also be clear, the avoidance so far of `@nest` in `dev-dot` is I think more related to us not being aware of it that anything else -  we definitely haven't discussed it as a team! It's quite possibly, likely even, that we have a ton of great use cases for `@nest`, and it seems like this could be one of them. Just nervous about introducing it into the project without that context of how & when we want to use it - but super open to it 🙇‍♂️

<details>
<summary>Random related thoughts, why I mentioned grouping</summary>

Something that's been bouncing around my mind is balancing strictness with regards to reducing `className` & JSX complexity vs strictness in grouping CSS by element (& maybe even pseudo-element?). As you mentioned the grouping even with the extra classNames is still the same - what sticks out to me about this is that we have to jump into a few different places within the CSS to figure out what's going on with the `::after` pseudo element across viewports (at left in screenshot below).

With this example, and the `.isFirstChild` & `.isLastChild` classNames in place, I think we could move things around a bit for more strict grouping... but also more lines of CSS, and the `border-radius: 14px` aren't as closely grouped...

![2022-04-14_0823](https://user-images.githubusercontent.com/4624598/163390629-038d6734-c949-469c-aa49-35d6617f0cd6.png)

I'm very on the fence about this though - can definitely see how `@nest` is preferable is a lotta ways 👍 But wanted to get this thought out there in case it sparks anything! 

</details>